File: Assignment 5/C/10. arrange a binary tree with a doubly ll .c
#include <stdio.h>
#include <stdlib.h>

typedef struct node{
    int val;
    struct node* left;
    struct node* right;
}node;

node* createNode(int n){
    node* cn = (node*)malloc(sizeof(node));
    cn->val = n;
    cn->left = NULL;
    cn->right = NULL;
    return cn;
}

void inOrderTrav(node* head){
    if(head==NULL){
        return;
    }
    inOrderTrav(head->left);
    printf("%d ",head->val);
    inOrderTrav(head->right);
}

int main(){
    node* head = createNode(1);
    node* snd = createNode(2);
    node* trd = createNode(3);
    node* four = createNode(4);
    node* five = createNode(5);
    
    head->left = snd;
    head->right = trd;

    snd->left = four;
    snd->right = five;

/*      1
       / \
      2   3
     / \
    4   5         */
    inOrderTrav(head);

    return 0;
}

File: Assignment 5/Python/10. arrange a binary tree with a doubly ll .py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def inOrderTrav(head):
    if head is None:
        return
    inOrderTrav(head.left)
    print(head.val, end=" ")
    inOrderTrav(head.right)

if __name__ == "__main__":
    head = Node(1)
    snd = Node(2)
    trd = Node(3)
    four = Node(4)
    five = Node(5)

    head.left = snd
    head.right = trd

    snd.left = four
    snd.right = five

    print("In-order traversal: ", end="")
    inOrderTrav(head)

# Output:
# In-order traversal: 4 2 5 1 3 
